### PR TITLE
IDR 0.3.1 merge branch

### PR DESCRIPTION
**Pull request by manics**
This is the current state of `demo31`.

Do not merge, this branch is split into multiple PRs.

### CODE REVIEW

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [MANAGEMENT_TOOLS-merge#1127](https://ci.openmicroscopy.org/job/MANAGEMENT_TOOLS-merge/1127/). See the [console output](https://ci.openmicroscopy.org/job/MANAGEMENT_TOOLS-merge/1127/consoleText) for more details.
Possible conflicts:
  - PR #220 manics 'Install JDK on idr OMERO for jstack (IDR-0.3.1)'
    - ansible/idr-playbooks/group_vars/omero-hosts.yml
  - PR #222 manics 'Bump IDR omero to 0.3.1-rc2, and idr-notebooks to HEAD (IDR-0.3.1)'
    - ansible/idr-playbooks/group_vars/omero-hosts.yml
  - PR #226 manics 'Update omero-web-apps role, update mapr (IDR-0.3.1)'
    - ansible/idr-playbooks/group_vars/omero-hosts.yml

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [MANAGEMENT_TOOLS-merge#1128](https://ci.openmicroscopy.org/job/MANAGEMENT_TOOLS-merge/1128/). See the [console output](https://ci.openmicroscopy.org/job/MANAGEMENT_TOOLS-merge/1128/consoleText) for more details.
Possible conflicts:
  - PR #220 manics 'Install JDK on idr OMERO for jstack (IDR-0.3.1)'
    - ansible/idr-playbooks/group_vars/omero-hosts.yml
  - PR #222 manics 'Bump IDR omero to 0.3.1-rc2, and idr-notebooks to HEAD (IDR-0.3.1)'
    - ansible/idr-playbooks/group_vars/omero-hosts.yml
  - PR #226 manics 'Update omero-web-apps role, update mapr (IDR-0.3.1)'
    - ansible/idr-playbooks/group_vars/omero-hosts.yml

**Comment by manics** [comment]
--exclude

File: ansible/roles/openstack-idr-instance/tasks/floatingip.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+---
+# Handle floating IPs with multiple NICs
+# The os_floating_ip module may behave in an unexpected manner.
+# These tasks attempt to work around them.
+# Even so you may still see problem with Ansible 2.1, if so try 2.2 or 2.3
+
+- name: idr vm | floating ip
```
Comment (joshmoore) [inline ansible/roles/openstack-idr-instance/tasks/floatingip.yml]: I'm currently having an issue with this:

```
The error appears to have been in '/Users/jamoore/opt/management_tools/infrastructure/ansible/roles/openstack-idr-instance/tasks/floatingip.yml': line 7, column 3, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:


- name: idr vm | floating ip
  ^ here
```

```
$ ansible --version
ansible 2.1.1.0
  config file = /Users/jamoore/opt/management_tools/idr/ansible.cfg
  configured module search path = Default w/o overrides
```

Comment (joshmoore) [inline ansible/roles/openstack-idr-instance/tasks/floatingip.yml]: Now blocked on testing the idr-0-3-1 branch due to this. Also with ansible 2.2.0.0

**Comment by manics** [comment]
Probably best to exclude fd3f48c97a384c3de523a2fd6af9b50e1c87a449 for now. All other commits should be in PRs.

**Comment by joshmoore** [comment]
Revert helped. I have a few local changes that have nearly gotten things running. Just looking into:

```
TASK [ice : zeroc ice | pip install packages] **********************************
failed: [joshA-omero] (item=zeroc-ice>=3.6,<3.7) => {"cmd": "/bin/pip2 install zeroc-ice>=3.6 <3.7", "failed": true, "item": "zeroc-ice>=3.6,<3.7", "msg": "\n:stderr: Invalid requirement: '<3.7'\nTraceback (most recent call last):\n  File \"/usr/lib/python2.7/site-packages/pip/req/req_install.py\", line 78, in __init__\n    req = Requirement(req)\n  File \"/usr/lib/python2.7/site-packages/pip/_vendor/packaging/requirements.py\", line 96, in __init__\n    requirement_string[e.loc:e.loc + 8]))\nInvalidRequirement: Invalid requirement, parse error at \"'<3.7'\"\n\nYou are using pip version 8.1.2, however version 9.0.1 is available.\nYou should consider upgrading via the 'pip install --upgrade pip' command.\n"}
```

Now.

**Comment by joshmoore** [comment]
And after some pip wrangling, now:

```
TASK [omero-server : omero | install omero] ************************************
fatal: [joshA-omero]: FAILED! => {"changed": false, "cmd": "/home/omero/omego/bin/omego install --release 0.3.1 --sym OMERO.server --prestartfile /home/omero/config/omero-base.config --prestartfile /home/omero/config/omero-additional.config --ice 3.6 -qq --downloadurl https://downloads.openmicroscopy.org/idr --no-start --dbhost 10.20.30.8 --dbuser omero --dbname idr --dbpass omero --rootpass omero --initdb", "failed": true, "msg": "[Errno 2] No such file or directory", "rc": 2}
```

**Comment by joshmoore** [comment]
omego looks to be installed globally (`/usr/bin/omego`)

**Comment by joshmoore** [comment]
Finally:

```
Dec 23 12:57:20 josha-proxy nginx[22835]: nginx: [emerg] BIO_new_file("/etc/nginx/ssl/idr-fullchain.pem") failed (SSL: error:02001002:system library:fopen:No such file or directory:fope...no such file)
```

**Comment by manics** [comment]
> omego looks to be installed globally (/usr/bin/omego)

That's an Ansible 2.2 bug, it creates the virtualenv but installs `omego` globally. I haven't managed to create a minimal working example yet

**Comment by manics** [comment]
> `failed: [joshA-omero] (item=zeroc-ice>=3.6,<3.7)`

Looks like https://github.com/openmicroscopy/infrastructure/pull/221 is missing

**Comment by joshmoore** [comment]
Ok....and those commits aren't on this branch. Check. How do you want to move forward here? Is there anything left here worth confirming? What do we do with https://github.com/manics/infrastructure/pull/4?

**Comment by manics** [comment]
Rebased.
- 6d59ea4 is https://github.com/openmicroscopy/infrastructure/pull/225
- The other two commits are ongoing work

**Pull request closed without merging**